dynamic_var.go

```diff
@@ -16,19 +16,24 @@ type dynamicFlag struct {
 
 func (df *dynamicFlag) Set(value string) error {
 	fieldKind := reflect.TypeOf(df.field).Elem().Kind()
-	var isBoolValue bool
-	if _, err := strconv.ParseBool(value); err == nil {
-		isBoolValue = true
-	}
-	if fieldKind == reflect.Bool && isBoolValue {
-		boolField := df.field.(*bool)
-		*boolField = true
-		return nil
+	var (
+		optionWithoutValue bool
+		pv                 bool
+	)
+	if value == "t" || value == "T" || value == "true" || value == "TRUE" {
+		pv = true
+		optionWithoutValue = true
+	} else if value == "f" || value == "F" || value == "false" || value == "FALSE" {
+		pv = false
 	}
+
 	switch fieldKind {
+	case reflect.Bool:
```

Comment:
We can leverage type abstractions instead of relying on reflection, e.g. https://github.com/projectdiscovery/utils/pull/266/files#diff-1e4c1fcda16b2f94c80249aa8e58bff941c8621fe28458d767fd3502c10c4f31R27